lib/net.js

```diff
@@ -413,28 +414,19 @@ Socket.prototype._final = function(cb) {
   const err = this._handle.shutdown(req);
 
   if (err === 1 || err === UV_ENOTCONN)  // synchronous finish
-    return afterShutdown.call(req, 0);
+    return cb();
   else if (err !== 0)
-    return this.destroy(errnoException(err, 'shutdown'));
+    return cb(errnoException(err, 'shutdown'));
 };
 
 
-function afterShutdown(status) {
+function afterShutdown() {
```

Comment:
@mcollina: Is the debug message important here? I would prefer to just skip `afterShutdown` and  just assign the callback directly without any indirection.